datafusion/core/tests/sqllogictests/README.md

```diff
@@ -97,17 +97,27 @@ Query records follow the format:
-- `expected_result`: In the results section, integer values are rendered as if by printf("%d"). Floating point values are rendered as if by printf("%.3f"). NULL values are rendered as "NULL". Empty strings are rendered as "(empty)". Within non-empty strings, all control characters and unprintable characters are rendered as "@".
-- `sort_mode`: If included, it must be one of "nosort", "rowsort", or "valuesort". The default is "nosort". In nosort mode, the results appear in exactly the order in which they were received from the database engine. The nosort mode should only be used on queries that have an ORDER BY clause or which only have a single row of result, since otherwise the order of results is undefined and might vary from one database engine to another. The "rowsort" mode gathers all output from the database engine then sorts it by rows on the client side. Sort comparisons use strcmp() on the rendered ASCII text representation of the values. Hence, "9" sorts after "10", not before. The "valuesort" mode works like rowsort except that it does not honor row groupings. Each individual result value is sorted on its own.
+- `type_string`: A short string that specifies the number of result columns and the expected datatype of each result column. There is one character in the <type_string> for each result column. The characters codes are:
+  - "T" for a text result,
+  - "I" for an integer result,
+  - "R" for a floating-point result,
+  - "?" for any other type.
+- `expected_result`: In the results section, some values are converted according to some rules:
+  - floating point values are rounded to the scale of "12",
+  - NULL values are rendered as `NULL`,
+  - empty strings are rendered as `(empty)`,
+  - boolean values are rendered as `true`/`false`,
+  - this list can be not exhaustive, check the `datafusion/core/tests/sqllogictests/src/engines/conversion.rs` for details.
+- `sort_mode`: If included, it must be one of `nosort`, `rowsort` (default), or `valuesort`. In `nosort` mode, the results appear in exactly the order in which they were received from the database engine. The `nosort` mode should only be used on queries that have an `ORDER BY` clause or which only have a single row of result, since otherwise the order of results is undefined and might vary from one database engine to another. The `rowsort` mode gathers all output from the database engine then sorts it by rows on the client side. Sort comparisons use [sort_unstable](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable) on the rendered text representation of the values. Hence, "9" sorts after "10", not before. The `valuesort` mode works like `rowsort` except that it does not honor row groupings. Each individual result value is sorted on its own.
```

Comment:
I like this idea to default to `rowsort` as it is less of a foot gun

Reply:
Oops, I actually made a mistake after looking at "rowsort" for a long time. 🤦 Sorry for the confusion.

`nosort` is the default. It has always been. An absence of the explicit ordering is the same as `nosort`. The default is also not configurable.

I fixed this doc https://github.com/apache/arrow-datafusion/pull/5110/commits/77d905d58d209aae35840dd9613a743f2f55a67a

Reply:
Regarding the default option, they are kind of the same.

`rowsort` by default would require adding "nosort+order by" in the order by cases. Forgetting `nosort` in this case leads to having a peculiar ordering for numbers, for example, because sorting is done on strings.

`nosort` by default requires adding `rowsort` when there is no order by. Forgetting `rowsort` leads to potentially unstable results.

I am not sure what's better. But anyway `nosort` is the default, and having any other defaults requires changes to `sqllogictest-rs`.

Sorry again for the confusion.